app/Routes.hs

```diff
@@ -3,13 +3,41 @@ module Routes
 
 import Control.Monad (MonadPlus (mplus), msum)
 import Control.Monad.IO.Class (liftIO)
+import Controllers.Course as CourseControllers (retrieveCourse, index, courseInfo, depts)
```

Comment:
Oh I guess I didn't pick up on this on your clarifying comment, but this should be `CoursesController` the entity name is pluralized, not "controller". Same with the graph controller below.